### ISSUE TITLE
Materials Printer

### ISSUE THREAD

**Issue by MaverickEsq** [d3athrow/vgstation13#24508]
Based on this PR https://github.com/vgstation-coders/vgstation13/pull/24223

I am far too shit to code this, but instead of having to order individual resources, a machine could be added to cargo that eats compressed matter cartridges and spits out stacks of the 4 basic materials (Wood/Metal/Plastic/Glass) as well as cable coils so we can stop ordering 50,000 electrical maintenence crates. It could also allow for simple blueprints like light floortile and reinforced glass to be added via scanning them.

Make the machine non-scannable by the device analyizer, but allow it to be moved just for the shits and giggles of shitfights after people steal the damn thing.

This adds the chore of ordering compressed matter cartridges and filling the materials printer to cargo's roundstart, but massively reduces the chore of having to deal with 40 crates that each have a single stack of metal in them when some asshole decides they want to build a spacebar. Possibly allow it to use dense matter from the roid like uranium as fuel also. Spritewise, it could be an organ printer but recolored.

[featurerequest] [qol]

**Comment by brndd** [comment]
It could be like a 3D printer of sorts. Basically an automatic lathe. An autolathe.

**Comment by utilitas-** [comment]
it's called a "matter synthesizer", retard
I'd much more appreciate someone making matter cartridges stackable so feeding the matter synthesizer is less painful

**Comment by MaverickEsq** [comment]
> it's called a "matter synthesizer", retard

Yeah if R&D ever get off their ass. This is a roundstart machine version, rather than some R&D weidlable


> It could be like a 3D printer of sorts. Basically an automatic lathe. An autolathe.

Yes, the Autolathe is where we go to get raw materials. _Briliant_

**Comment by brndd** [comment]
No but get this

- Add cable coils to the autolathe

- Allow the autolathe to recycle compressed matter cartridges (if it can't already) to produce whatever the fuck raw materials you want

The second point would require tweaking compressed matter cartridges, because IIRC they can be made at the autolathe for some large amount of metal and that's no good if you want them to be recycled into metal+glass+plastic+wood (transmutation is a sin). Either change their recipe or add some dumb new omnimatter cartridge whose only purpose in life is to be recycled into all 4 types.

Though I don't really see what difference it makes to order matter cartridges to turn into all 4 materials vs. just ordering said materials directly when you need them.

**Comment by utilitas-** [comment]
you motherfuckers gotta overcomplicate everything
how 'bout this
slap an autolathe with the matter synthesizer from RnD to unlock material synthesis capability (that, or fully upgrading it to super tier components, but that may be too easy)
having a free magical matter synthesis machine from roundstart would completely invalidate all of mining after the first batch of minerals (if even that, considering you can make everything in xenobio)
and the only reason I can come up with for why the autolathe shouldn't _just be able to create cable coils already_ is that cable coils are a stack, and that might be a bit wonky since it'd have to print it wire by wire if it doesn't shit it out by the stackload
but I'd be okay with it if printing cables required silver, at least it'd give the mineral some use (plus, recycling/grinding cables in a mortar would give you teeny bits of silver which could be useful for some ghetto chem)

**Comment by MaverickEsq** [comment]
Material synthesis is uranium, plasma, phazon, banananinium. I think having an uncopyable machine in cargo (where one can order stacks of wood, metal, plastic and glass) isn't overcomplicating shit. Its just meaning having to rely less on cargo when you wanna build shit. And encouraging building shit isn't such a bad idea. Fuck, make it need money to print metal, same thing as cargo but without having to speak to some vox motherfucker who can't seem to grasp the concept of "call me to come get my shit, don't just give it to some random greyshit who says hes here for metal"

Besides, mining is only useful for plasma glass, uranium and gold for R&D. Who the fuck gets metal from roid.

**Comment by utilitas-** [comment]
having ANY uncopiable anything for the sake of it is overcomplicating shit. you're bloating the game with pointless machinery that slowly but surely leads to fuckhuge stations full of empty space. just look at the kitchen: half of the machinery in there gets used 1% of the time, maybe, and most of it hasn't even been ported to true machinery code. then there's most of the research outpost, or the gateway, et cetera. also
\>rely less on cargo
\>let's give cargo a unique snowflake machine that cannot be copied (one among approximately two (2) others in the entire game)
and literally everyone gets metal from the roid, where the fuck were you? last 5 rounds I've played metal made up the majority of any haul from mining simply because mining does not pay at all anymore (thanks a lot kurfurst)
